apps/nextjs-website/src/components/organisms/WebinarsTemplate/WebinarsTemplate.tsx

```diff
@@ -0,0 +1,92 @@
+'use client';
+import React, { useEffect, useState } from 'react';
+import Hero from '@/editorialComponents/Hero/Hero';
+import { useTranslations } from 'next-intl';
+import { Box, Grid, useTheme } from '@mui/material';
+import { Webinar } from '@/lib/types/webinar';
+import EContainer from '@/editorialComponents/EContainer/EContainer';
+import SectionTitle from '@/components/molecules/SectionTitle/SectionTitle';
+import WebinarsSection from '@/components/organisms/WebinarsSection/WebinarsSection';
+import WebinarListItem from '@/components/molecules/WebinarListItem/WebinarListItem';
+
+const CHECK_WEBINARS_INTERVAL_MS = 60 * 1000;
+
+type WebinarsTemplateProps = {
+  webinars: readonly Webinar[];
+};
+
+const WebinarsTemplate = ({ webinars }: WebinarsTemplateProps) => {
+  const t = useTranslations();
+  const { palette } = useTheme();
+  const [nextWebinars, setNextWebinars] = useState<Webinar[]>([]);
+  const [pastWebinars, setPastWebinars] = useState<Webinar[]>([]);
+
+  const getNextWebinars = (currentWebinars: readonly Webinar[]) =>
+    currentWebinars.filter(
+      ({ startDateTime, endDateTime }) =>
+        (!startDateTime && !endDateTime) ||
+        (endDateTime && new Date(endDateTime).getTime() > new Date().getTime())
+    );
+
+  const getPastWebinars = (currentWebinars: readonly Webinar[]) =>
+    currentWebinars
+      .filter(
+        ({ endDateTime, startDateTime }) =>
+          endDateTime &&
+          startDateTime &&
+          new Date(endDateTime).getTime() <= new Date().getTime()
+      )
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      .sort((a, b) => (a.startDateTime! < b.startDateTime! ? 1 : -1));
```

Comment:
I'd move these 2 functions outside the component, or wrap them in a `useCallback` hook so they are not redefined every re-render of the component

Reply:
When using `useCallback` the current date never changes. I think it's better to leave the code as it is

Reply:
Done